modules/framework/galasa-parent/dev.galasa.framework.resource.management/src/main/java/dev/galasa/framework/resource/management/internal/MonitorFilter.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.resource.management.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+import dev.galasa.framework.spi.FrameworkException;
+
+public class MonitorFilter {
+    
+    private List<Pattern> includes;
+    private List<Pattern> excludes;
+
+    public MonitorFilter(List<String> includesGlobList, List<String> excludesGlobList) throws FrameworkException {
+        this.includes = convertGlobListToPatternList(includesGlobList);
+        this.excludes = convertGlobListToPatternList(excludesGlobList);
+    }
+
+    public boolean isMonitorClassAllowed(String monitorClassName) {
+        boolean isAllowed = (
+            isStringMatchingAnyPatternInList(monitorClassName, includes) && !isStringMatchingAnyPatternInList(monitorClassName, excludes)
+        );
+
+        return isAllowed;
+    }
+
+    private boolean isStringMatchingAnyPatternInList(String strToCheck, List<Pattern> patterns) {
+        boolean isMatching = false;
+
+        for (Pattern pattern : patterns) {
+            Matcher patternMatcher = pattern.matcher(strToCheck);
+            if (patternMatcher.matches()) {
+                isMatching = true;
+                break;
+            }
+        }
+        return isMatching;
+    }
+
+    private List<Pattern> convertGlobListToPatternList(List<String> globList) throws FrameworkException {
+        List<Pattern> patternList = new ArrayList<>();
+        for (String globPattern : globList) {            
+            try {
+                String convertedPattern = getGlobAsRegexString(globPattern);
+                Pattern patternToAdd = Pattern.compile(convertedPattern);
+                patternList.add(patternToAdd);
+            } catch (PatternSyntaxException e) {
+                throw new FrameworkException("Failed to compile glob pattern into a valid regex pattern", e);
+            }
+        }
+        return patternList;
+    }
+
+    private String getGlobAsRegexString(String globPattern) {
```

Comment:
JavaClassGlobPattern class needs to process a single glob, turn to regex, fail to construct if it's invalid...etc.

Reply:
only supporting * and ? excluding any characters which are not fully qualified java class names

Reply:
Addressed in https://github.com/galasa-dev/galasa/pull/236/commits/a9ef197a47c1c7da97f4c1566f80ca24bce91184